File: Clojure/src/com/yourtion/Pattern01/person_example.clj
(ns com.yourtion.Pattern01.person-example)

(def p1 {:first-name "Michael" :last-name "Bevilacqua"})
(def p2 {:first-name "Pedro" :last-name "Vasquez"})
(def p3 {:first-name "Robert" :last-name "Aarons"})

(def people [p1 p2 p3])


(def sorted_people (sort (fn [p1 p2] (compare (p1 :last-name) (p2 :last-name))) people))

;(sort (fn [p1 p2] (compare (p1 :first-name) (p2 :first-name))) people)

(defn run [] (doseq [col sorted_people :when (not= col \B)]
            (println (col :first-name) (col :last-name))))

File: Scala/src/com/yourtion/Pattern01/PersonExample.scala
package com.yourtion.Pattern01

/**
  * Created by Yourtion on 9/7/16.
  */
object PersonExample {
  (int1: Int, int2: Int) => int1 + int2

  case class Person(firstName: String, lastName: String)

  val p1 = Person("Michael", "Bevilacqua")
  val p2 = Person("Pedro", "Vasquez")
  val p3 = Person("Robert", "Aarons")

  val people = Vector(p3, p2, p1)

  val sorted_people = people.sortWith((p1, p2) => p1.firstName < p2.firstName)

  def run() = {
    sorted_people.foreach(p => println(p.firstName, p.lastName))
  }
}
